compiler/src/main/java/org/neo4j/ogm/compiler/builders/DefaultNodeBuilder.java

```diff
@@ -25,39 +25,52 @@
  */
 public class DefaultNodeBuilder implements NodeBuilder {
 
-	NodeModel node = new NodeModel();
-
-
-	public DefaultNodeBuilder(Long reference) {
-		node.setId(reference);
-	}
-
-	@Override
-	public NodeBuilder addProperty(String key, Object value) {
-		node.getPropertyList().add(new PropertyModel<>(key, value));
-		return this;
-	}
-
-	@Override
-	public NodeBuilder setLabels(Collection<String> newLabels) {
-		String[] labels = new String[newLabels.size()];
-		labels = newLabels.toArray(labels);
-		node.setLabels(labels);
-		return this;
-	}
-
-	@Override
-	public Long reference() {
-		return node.getId();
-	}
-
-	@Override
-	public String[] labels() {
-		return node.getLabels();
-	}
-
-	@Override
-	public Node node() {
-		return node;
-	}
+    NodeModel node = new NodeModel();
+
+
+    public DefaultNodeBuilder(Long reference) {
+        node.setId(reference);
+    }
+
+    @Override
+    public NodeBuilder addProperty(String key, Object value) {
+        node.getPropertyList().add(new PropertyModel<>(key, value));
+        return this;
+    }
+
+    @Override
+    public NodeBuilder addLabels(Collection<String> newLabels) {
+        String[] labels;
+        labels = newLabels.toArray(new String[newLabels.size()]);
+        node.setLabels(labels);
+        return this;
+    }
+
+    @Override
+    public Long reference() {
+        return node.getId();
+    }
+
+    @Override
+    public String[] addedLabels() {
+        return node.getLabels();
+    }
+
+    @Override
```

Comment:
I'd prefer to keep `labels` as the property representing the current set of labels for the node.